python/ray/_private/runtime_env/pip.py

```diff
@@ -86,6 +92,57 @@ def _is_in_virtualenv() -> bool:
             hasattr(sys, "base_prefix") and sys.base_prefix != sys.prefix
         )
 
+    @classmethod
+    async def _ensure_pip_version(
+        cls,
+        path: str,
+        pip_version: Optional[str],
+        cwd: str,
+        pip_env: Dict,
+        logger: logging.Logger,
+    ):
+        if not pip_version:
+            return
+
+        python = _PathHelper.get_virtualenv_python(path)
+        # Ensure pip version.
+        pip_reinstall_cmd = [
+            python,
+            "-m",
+            "pip",
+            "install",
+            "--disable-pip-version-check",
+            f"pip{pip_version}",
+        ]
+        logger.info("Installing pip version to %s", pip_version)
+
+        await check_output_cmd(pip_reinstall_cmd, logger=logger, cwd=cwd, env=pip_env)
+
+    async def _pip_check(
+        self,
+        path: str,
+        pip_check_enable: bool,
+        cwd: str,
+        pip_env: Dict,
+        logger: logging.Logger,
+    ):
+        if not pip_check_enable:
+            logger.info("Skip pip check.")
+            return
+        python = _PathHelper.get_virtualenv_python(path)
+
+        output = await check_output_cmd(
+            [python, "-m", "pip", "check", "--disable-pip-version-check"],
```

Comment:
This is confusing. Why do we disable pip version check when pip check is enabled?